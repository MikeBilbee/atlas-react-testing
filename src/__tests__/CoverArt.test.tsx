//CoverArt.test.tsx
import { describe, it, expect } from 'vitest';
import { render } from '@testing-library/react';
import CoverArt from '../components/CoverArt';

describe('CoverArt component', () => {
  it('renders with provided imageUrl', () => {
    const imageUrl = 'https://example.com/image.jpg';
    const { container } = render(<CoverArt cover={imageUrl} />);
    expect(container).toMatchSnapshot();
  });

  it('renders with placeholder image when imageUrl is not provided', () => {
    const { container } = render(<CoverArt />);
    expect(container).toMatchSnapshot();
  });
});