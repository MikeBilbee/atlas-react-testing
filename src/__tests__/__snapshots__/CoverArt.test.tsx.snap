// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CoverArt component > renders with placeholder image when imageUrl is not provided 1`] = `
<div>
  <div
    class="w-30 h-30 overflow-hidden rounded-lg shadow-md"
  >
    <img
      alt="Cover Art"
      class="w-full h-full object-cover"
      src="/src/assets/placeholder.svg"
    />
  </div>
</div>
`;

exports[`CoverArt component > renders with provided imageUrl 1`] = `
<div>
  <div
    class="w-30 h-30 overflow-hidden rounded-lg shadow-md"
  >
    <img
      alt="Cover Art"
      class="w-full h-full object-cover"
      src="https://example.com/image.jpg"
    />
  </div>
</div>
`;
